test/jdk/java/net/httpclient/lib/jdk/httpclient/test/lib/common/HttpServerAdapters.java

```diff
@@ -421,6 +425,50 @@ private void doHandle(HttpTestExchange t) throws IOException {
         }
     }
 
+    /**
+     * An {@link HttpTestHandler} that handles only HEAD and GET
+     * requests. If another method is used 405 is returned with
+     * an empty body.
+     * The response is always returned with fixed length.
+     */
+    public static class HttpHeadHandler implements HttpTestHandler {
```

Comment:
The class name is slightly misleading since this also handles `GET`. Perhaps `HeadOrGetHandler` would be appropriate?

Reply:
And maybe we should make this a top level public class in this test package to reduce the amount of code we have in the `HttpServerAdapters` file.

Reply:
Renamed HttpHeadOrGetHandler. I'd rather keep the implementation in this file. We already have an HttpEchoHandler here, and there are two many classes bearing this name in the test class hierarchy to consider making it top level.